docs/source/community/rfc/rfc-2.rst

```diff
@@ -0,0 +1,904 @@
+
+.. _`iso19111 branch of rouault/proj.4 repository`: https://github.com/rouault/proj.4/tree/iso19111
+
+.. _`comparison with a fixed snapshot of master branch`: https://github.com/OSGeo/proj.4/compare/iso19111_dev...rouault:iso19111
+
+
+Database
+--------
+
+Content
+*******
+
+The database contains CRS and coordinate operation definitions from the EPSG
+database v9.5.3 and the IGNF registry, as well as a few customizations.
+
+Building (for PROJ developers creating the database)
+****************************************************
+
+The building of the database is a several stage process:
+
+    - the first stage consists in constructing .sql scripts mostly with
+      CREATE TABLE and INSERT statements to create the database structure and
+      populate it. There is one .sql file for each database table, populated
+      with the content of the EPSG database, automatically
+      generated with the `build_db.py`_ script, which processes the PostgreSQL
+      dumps issued by IOGP. A number of other scripts are dedicated to manual
+      edition, for example `grid_alternatives.sql`_ file that binds official
+      grid names to PROJ grid names
+
+    - the second stage is done automatically by the make process. It pipes the
+      .sql script, in the right order, to the sqlite3 binary to generate a
+      first version of the proj.db SQLite3 database.
+
+    - the third stage consists in creating additional .sql files from the
+      content of other registries. For that process, we need to bind some
+      definitions of those registries to those of the EPSG database, to be
+      able to link to existing objects and detect some boring duplicates.
+      The `ignf.sql`_ file has been generated using
+      the `build_db_create_ignf.py`_ script from the current data/IGNF file
+      that contains CRS definitions (and implicit transformations to WGS84)
+      as PROJ.4 strings.
+      The `esri.sql`_ file has been generated using the `build_db_from_esri.py`_
+      script, from the .csv files in
+      https://github.com/Esri/projection-engine-db-doc/tree/master/csv
+
+    - the last stage runs make again to incorporate the new .sql files generated
+      in the previous stage (so the process of building the database involves
+      a kind of bootstrapping...)
+
+Building (for PROJ users)
+*************************
+
+The make process just runs the second stage mentionned above from the .sql
+files. The resulting proj.db is currently 5.3 MB large.
+
+Structure
+*********
+
+The database is structured into the following tables and views. They generally
+match a ISO-19111 concept, and is generally close to the general structure of
+the EPSG database. Regarding identification of objects, where the EPSG database
+only contains a 'code' numeric column, the PROJ database identifies objects
+with a (auth_name, code) tuple of string values, allowing several registries to
+be combined together.
+
```

Comment:
This is not actually true. authority_list is a view, but perhaps I should rename it as _view

Reply:
I wasn't thinking about the names of the table or view (just thinking of the text that follows), but now you have me wondering what's better.  I don't see a reason to name it `_view`.  Doesn't really matter, yes?  Be it table or view, I believe it works the same from a query perspective.  Or go all out and have fun: `authority_list_table_view_builder_generator_non_authorative_catalog_oh_my_brain`

Reply:
The initial reason for _view was that I had a "coordinate_operation" table (that just listed the auth_name, code, table_name) and that served as a foreign key target for the "helmert_transformation", "grid_transformation", etc. table so that there is no duplication of (auth_name, code) between those "derived" tables. The coordinate_operation_view was actually a view selecting in those "derived" tables and having much more information.
I've finally dropped the coordinate_operation table since that was a bit painful and not user friendly, and replaced it with trigger checks.
I'm not yet sure if the _view naming convention is better or not. It has the advantage of indicating you cannot insert into it.